### ISSUE TITLE
'buffalo test' requires a database, can we remove it?

### ISSUE THREAD

**Issue by arschles** [gomods/athens#450]
`buffalo test` expects one of the supported RDBMS's to be running so that it can set up tables and schemas, etc... before running `go test`. If we remove RDBMS support from the proxy (#423), then we won't need that functionality, and it'll actually get in the way of running tests because you shouldn't have to run an unused database just to run tests

**Comment by michalpristas** [comment]
can we close in favor of #494 ?

**Comment by manugupt1** [comment]
Not sure, we can run buffalo test and see if it calls setup and teardown of databases. Want to check this for you this evening?

**Comment by michalpristas** [comment]
yeah let's try it and in case it blows let's file a bug on buffalo

**Comment by manugupt1** [comment]
This is the PR that will really let buffalo test not fail.
https://github.com/gomods/athens/pull/511

The relevant commits are : https://github.com/gomods/athens/pull/511/files#diff-f47e73f93c08e297a441f8902cc5c34e


Sorry about this big PR :(


It has one more change remaining and hopefully everything will work then!
